src/test/kotlin/ecommerce/ProductControllerTest.kt

```diff
@@ -0,0 +1,133 @@
+        )
+
+        val splitUpAttributes: List<Array<String>> =
+            listOf(
+                "Coca-Cola 2.00 https://mcdonalds.com.mt/wp-content/uploads/2024/06/" +
+                    "COCA-COLA-WEBSITE-IMG.jpg",
+                "Fanta 2.50 https://www.cokesolutions.com/content/dam/cokesolutions/us/" +
+                    "images/Products/Fanta-Orange-PET.jpg",
+                "Cappuccino 4.39 https://www.tchibo.de/kaffeeakademie/media/pages/global-images/" +
+                    "fb95bb5370-1729609446/adobestock_219364830-1440x700-crop-42-46.jpg",
+                "Tea 1.59 https://cupitol.com/wp-content/uploads/2019/08/tea-drinking-1.jpg",
+            ).map { name -> name.split(" ").toTypedArray() }.toList()
+        jdbcTemplate.batchUpdate("INSERT INTO products(name, price, image_url) VALUES (?,?,?)", splitUpAttributes)
+    }
+
+    @Test
+    fun create() {
+        val response =
+            RestAssured.given()
+                .log().all()
+                .contentType(ContentType.JSON)
+                .body(Product(name = "cola", price = 4.50, imageUrl = "https://cola.jpg"))
+                .`when`()
+                .post("/products")
+                .then()
+                .log().all()
+                .extract()
+
+        assertThat(response.statusCode()).isEqualTo(HttpStatus.CREATED.value())
+    }
+
+    @Test
+    fun read() {
+        val products = productRepository.findAllProducts()
+        assertThat(products.size).isEqualTo(4)
+        with(products[0]) {
+            assertThat(id).isEqualTo(1)
+            assertThat(name).isEqualTo("Coca-Cola")
+            assertThat(price).isEqualTo(2.00)
+            assertThat(imageUrl).isEqualTo("https://mcdonalds.com.mt/wp-content/uploads/2024/06/COCA-COLA-WEBSITE-IMG.jpg")
+        }
+        with(products[1]) {
+            assertThat(id).isEqualTo(2)
+            assertThat(name).isEqualTo("Fanta")
+            assertThat(price).isEqualTo(2.50)
+            assertThat(
+                imageUrl,
+            ).isEqualTo(
+                "https://www.cokesolutions.com/content/dam/cokesolutions/us/images/Products/" +
+                    "Fanta-Orange-PET.jpg",
+            )
+        }
+        with(products[2]) {
+            assertThat(id).isEqualTo(3)
+            assertThat(name).isEqualTo("Cappuccino")
+            assertThat(price).isEqualTo(4.39)
+            assertThat(
+                imageUrl,
+            ).isEqualTo(
+                "https://www.tchibo.de/kaffeeakademie/media/pages/global-images/fb95bb5370-1729609446/" +
+                    "adobestock_219364830-1440x700-crop-42-46.jpg",
+            )
+        }
+        with(products[3]) {
+            assertThat(id).isEqualTo(4)
+            assertThat(name).isEqualTo("Tea")
+            assertThat(price).isEqualTo(1.59)
+            assertThat(imageUrl).isEqualTo("https://cupitol.com/wp-content/uploads/2019/08/tea-drinking-1.jpg")
+        }
+    }
```

Comment:
It looks good to override the `equals` and `hashCode` methods to compare objects!
(Reference: `data class`) 😊